### PR TITLE
update dependencies and add connect icon animation

### PR DESCRIPTION
**Pull request by mixmasala**

### CODE REVIEW

**Comment by mixmasala** [comment]
connect icon animation should be refactored to eliminate data race on the connecting icon index shared between home.go and spool.go

File: home.go [hunk 1]
```diff
@@ -324,6 +331,17 @@ func (p *HomePage) Event(gtx layout.Context) interface{} {
 }
 
 func (p *HomePage) Start(stop <-chan struct{}) {
+	go func() {
+		for {
+			connectIconIdx = (connectIconIdx + 1) % len(connectIcons)
```
Comment (threebithacker) [inline home.go:336]: Is this necesssary here when we already do it in `SpoolPage.Start` (where it actually gets drawn?)

File: spool.go [hunk 2]
```diff
@@ -60,6 +61,8 @@ func (p *SpoolPage) Layout(gtx layout.Context) layout.Dimensions {
 					func() layout.FlexChild {
 						if isConnected {
 							return layout.Rigid(button(th, p.connect, connectIcon).Layout)
+						} else if isConnecting {
+							return layout.Rigid(button(th, p.connect, connectIcons[connectIconIdx]).Layout)
 						}
 						return layout.Rigid(button(th, p.connect, disconnectIcon).Layout)
 					}(),
```
Comment (threebithacker) [inline spool.go:68]: what's the point of using an anonymous function call here instead of a switch-case on the line before `return layout.Flex{Axis..` ?

**Review by threebithacker — CHANGES_REQUESTED** [review]

File: spool.go (hunk 2 above)
Comment (mixmasala) [reply spool.go]: oh man this pr is messed up. sorry. i pushed some different code and none of this exists any more. sorry.

**Pull request closed without merging**